backend/remote-state/s3/backend_test.go

```diff
@@ -29,16 +29,12 @@ func TestBackend_impl(t *testing.T) {
 }
 
 func TestBackendConfig(t *testing.T) {
-	// This test just instantiates the client. Shouldn't make any actual
-	// requests nor incur any costs.
-
+	testACC(t)
```

Comment:
Is this change because this test now *does* make actual requests? I assume that's requests to e.g. the `sts` APIs, the EC2 metadata, etc?

(If so, it could be helpful to have a comment here clarifying that, so those running the tests know what's required of them. Perhaps it could just say "this has the same requirements as the AWS provider acceptance tests".)